src/interpreter.rs

```diff
@@ -81,6 +94,47 @@ impl Default for Interpreter {
     }
 }
 
+impl Clone for Interpreter {
+    fn clone(&self) -> Self {
+        Self {
+            compiled_policy: self.compiled_policy.clone(),
+
+            data: self.data.clone(),
+            init_data: self.init_data.clone(),
+            input: self.input.clone(),
+            with_document: self.with_document.clone(),
+            with_functions: self.with_functions.clone(),
+
+            gather_prints: self.gather_prints,
+            prints: self.prints.clone(),
+            strict_builtin_errors: self.strict_builtin_errors,
+            traces: self.traces.clone(),
+
+            extensions: self.extensions.clone(),
+
+            #[cfg(feature = "coverage")]
+            coverage: self.coverage.clone(),
+            #[cfg(feature = "coverage")]
+            enable_coverage: self.enable_coverage,
+
+            // The following fields always get cleared during an evaluation.
+            // Hence, they need not be copied.
```

Comment:
is this why you need to implement the Clone function yourself rather than using the #derive(Clone)?

Reply:
also, when/where do we clone the interpreter?

Reply:
> is this why you need to implement the Clone function yourself rather than using the #derive(Clone)?

Yes.

Reply:
> also, when/where do we clone the interpreter?

Interpreter is held on to by the Engine and gets cloned when the Engine is cloned. The Interpreter's clone will share the CompiledPolicy with the instances from which it was cloned.

Regorus started off with the Interpreter being the core entity and later Engine was added.
We need to refactor and clean up these concepts - the Interpreter could ideally be split up.